lib/cadet/assessments/assessments.ex

```diff
@@ -3,29 +3,132 @@ defmodule Cadet.Assessments do
   Assessments context contains domain logic for assessments management such as
   missions, sidequests, paths, etc.
   """
-  use Cadet, :context
+  use Cadet, [:context, :display]
 
-  alias Cadet.Assessments.QuestionTypes.MCQQuestion
-  alias Cadet.Assessments.QuestionTypes.ProgrammingQuestion
+  import Ecto.Query
 
-  # To be uncommented when assessments context is merged
+  alias Timex.Duration
+
+  alias Cadet.Assessments.Assessment
+  alias Cadet.Assessments.Question
+
+  def all_assessments() do
+    Repo.all(Assessment)
+  end
+
+  def all_assessments(category) do
+    Repo.all(from(a in Assessment, where: a.category == ^category))
+  end
+
+  def all_open_assessments(category) do
+    now = Timex.now()
+
+    assessment_with_category = Repo.all(from(a in Assessment, where: a.category == ^category))
+    Enum.filter(assessment_with_category, &(&1.is_published and Timex.before?(&1.open_at, now)))
+  end
+
+  def assessments_due_soon() do
+    now = Timex.now()
+    week_after = Timex.add(now, Duration.from_weeks(1))
+
+    all_assessments()
+    |> Enum.filter(
+      &(&1.is_published and Timex.before?(&1.open_at, now) and
+          Timex.between?(&1.close_at, now, week_after))
+    )
+  end
+
+  def build_assessment(params) do
+    Assessment.changeset(%Assessment{}, params)
+  end
+
+  def build_question(params) do
+    Question.changeset(%Question{}, params)
+  end
+
+  def create_assessment(params) do
+    changeset = build_assessment(params)
```

Comment:
Use pipeline syntax

Reply:
https://github.com/lexmag/elixir-style-guide#needless-pipeline
Basically for single function call, don't use pipeline syntax.

Reply:
i was thinking of 
```
params
|> build_assessment()
|> Repo.insert()
```

Reply:
Ohh makes sense